Services/Statistics/SimulationStatisticsAgent.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Clustering;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Runtime;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Storage;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Statistics
+{
+    public interface ISimulationStatistics
+    {
+        Task<SimulationStatisticsModel> GetSimulationStatisticsAsync(string simulationId);
+        Task CreateOrUpdateAsync(string simulationId, SimulationStatisticsModel statistics);
+    }
+
+    public class SimulationStatisticsAgent : ISimulationStatistics
+    {
+        private readonly IStorageRecords simulationStatisticsStorage;
+        private readonly IClusterNodes clusterNodes;
+        private readonly ISimulations simulations;
+        private readonly IServicesConfig config;
+        private readonly ILogger log;
+
+        public SimulationStatisticsAgent(IServicesConfig config,
+            IClusterNodes clusterNodes,
+            ISimulations simulations,
+            IFactory factory,
+            ILogger logger)
+        {
+            this.clusterNodes = clusterNodes;
+            this.simulations = simulations;
+            this.config = config;
+            this.log = logger;
+            this.simulationStatisticsStorage = factory.Resolve<IStorageRecords>().Init(config.StatisticsStorage);
+        }
+
+        public async Task<SimulationStatisticsModel> GetSimulationStatisticsAsync(string simulationId)
+        {
+            string sqlCondition = " CONTAINS(ROOT.id, @simulationId)";
```

Comment:
I don't know if you're validating the input in the caller, but, if not, you should do some input validation here.